File: src/test/java/com/snackgame/server/member/domain/GuestNameRandomizerTest.java
package com.snackgame.server.member.domain;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

@SuppressWarnings("NonAsciiCharacters")
@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
class GuestNameRandomizerTest {

    private final GuestNameRandomizer nameRandomizer = new GuestNameRandomizer();

    @Test
    void 이름앞에_접두사가_붙는다() {
        var name = nameRandomizer.get().getString();

        assertThat(name).startsWith("게스트#");
    }

    @Test
    void 이름의_8자리를_무작위로_생성한다() {
        var name = nameRandomizer.get().getString();

        assertThat(name.substring(4)).hasSize(8);
    }
}

File: src/test/java/com/snackgame/server/member/service/GuestNameRandomizerTest.kt
@file:Suppress("NonAsciiCharacters")

package com.snackgame.server.member.service

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class GuestNameRandomizerTest {

    private val guestNameRandomizer = GuestNameRandomizer()
    private val randomizedLength = 7

    @Test
    fun `이름 앞에 접두사 및 '_' 가 붙는다`() {
        val randomized = guestNameRandomizer.getWith("guest").string

        assertThat(randomized).startsWith("guest_")
    }

    @Test
    fun `'_' 뒤 알파벳 7자리를 무작위로 생성한다`() {
        val randomized = guestNameRandomizer.getWith("guest").string

        assertThat(randomized.substringAfter('_')).hasSize(randomizedLength)
    }
}
